src/main/java/com/iheartradio/m3u8/MediaPlaylistLineParser.java

```diff
@@ -285,6 +283,31 @@ public void parse(String line, ParseState state) throws ParseException {
         }
     };
 
+    static final IExtTagParser EXT_X_BYTE_RANGE = new IExtTagParser() {
+        private final LineParser lineParser = new MediaPlaylistLineParser(this);
+
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_BYTE_RANGE_TAG;
+        }
+
+        @Override
+        public boolean hasData() {
+            return true;
+        }
+
+        @Override
+        public void parse(String line, ParseState state) throws ParseException {
+            lineParser.parse(line, state);
+            final Matcher matcher = ParseUtil.match(Constants.EXT_X_BYTE_RANGE_PATTERN, line, getTag());
+
+            if (matcher.groupCount() == 3) {
```

Comment:
I do not believe there is a case where the group count can be anything other than the total number of groups. Even if that group is empty, group count should return the same number. I think we can remove the `if` condition and keep the code in the block since it's guaranteed to have a match after the call to `match` above.